src/main/java/com/yahoo/bullet/parsing/expressions/LazyExpression.java

```diff
@@ -0,0 +1,133 @@
+package com.yahoo.bullet.parsing.expressions;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import com.yahoo.bullet.common.Configurable;
+import com.yahoo.bullet.common.Initializable;
+import com.yahoo.bullet.typesystem.Type;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Use "Lazy" as a placeholder name because it rolls off the tongue.
```

Comment:
.... You know what's better than things rolling off the tongue? Not rolling anything off the tongue. Just omit the Lazy... 🎉